### PR TITLE
AMQP-466: Reinstate and Deprecate Constant

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-466

Avoid (unlikely, but possible) breaking change.

__cherry-pick to 1.4.x__

### CODE REVIEW

File: spring-amqp/src/main/java/org/springframework/amqp/core/AddressUtils.java [hunk 1]
```diff
@@ -23,6 +23,12 @@
 public class AddressUtils {
 
 	/**
+	 * @deprecated Use the constant in {@link Address}.
+	 */
+	@Deprecated
+	public static final String AMQ_RABBITMQ_REPLY_TO = "amq.rabbitmq.reply-to";
```
Comment (artembilan) [inline spring-amqp/src/main/java/org/springframework/amqp/core/AddressUtils.java]: `AMQ_RABBITMQ_REPLY_TO = Address.AMQ_RABBITMQ_REPLY_TO;`
:smile:

**Comment by artembilan** [comment]
And I guess: this `AddressUtils` can be `@Deprecated` altogether

File: spring-amqp/src/main/java/org/springframework/amqp/core/AddressUtils.java (hunk 1 above)
Comment (garyrussell) [inline spring-amqp/src/main/java/org/springframework/amqp/core/AddressUtils.java]: Doh - can you fix?

Comment (artembilan) [inline spring-amqp/src/main/java/org/springframework/amqp/core/AddressUtils.java]: Sure. merging...

**Comment by artembilan** [comment]
Something wrong was with me this morning.
It strange that I haven't done that on the first review...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/5c1be6c6c020851b143978f618a97116101b3eb3

**Comment by artembilan** [comment]
... and cherry-picked

**Pull request closed without merging**